hugolib/author.go

```diff
@@ -13,23 +13,57 @@
 
 package hugolib
 
-// AuthorList is a list of all authors and their metadata.
-type AuthorList map[string]Author
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cast"
+)
+
+var (
+	onlyNumbersRegExp = regexp.MustCompile("^[0-9]*$")
+)
+
+// Authors is a list of all authors and their metadata.
+type Authors []Author
+
+// Get returns an author from an ID
+func (a Authors) Get(id string) Author {
+	for _, author := range a {
+		if author.ID == id {
+			return author
+		}
+	}
+	return Author{}
+}
+
+// Sort sorts the authors by weight
+func (a Authors) Sort() Authors {
```

Comment:
This isn't needed. The authors provided by node and page should already be sorted.